File: 745-find-smallest-letter-greater-than-target/find-smallest-letter-greater-than-target.py
class Solution:
    def nextGreatestLetter(self, letters: List[str], target: str) -> str:
        n  = len(letters)
        for i in range(n):
            c = letters[i]
            if c <= target:
                i+=1
            else:
                break
        return letters[i] if i<n else letters[0]
        

File: 745-find-smallest-letter-greater-than-target/find-smallest-letter-greater-than-target.java
import java.util.Arrays;

class Solution {
    public char nextGreatestLetter(char[] letters, char target) {
        int idx = Arrays.binarySearch(letters, target);

        // If found, skip all duplicates
        if (idx >= 0) {
            while (idx < letters.length && letters[idx] == target) {
                idx++;
            }
        } else {
            // target not found → get insertion point
            idx = -idx - 1;
        }

        // wrap-around if target >= all letters
        return letters[idx % letters.length];
    }
}

